### PR TITLE
Various fixes for Simplelink/sensortag sensors 

### PR DESCRIPTION
**Pull request by g-oikonomou**
This PR starts addressing some of the sensortag sensor flaws identified for platform Simplelink in #1282

* We fix an error in a CPP conditional which resulted in the platform startup code failing to initialise the sensors module correctly.
* We fix the I2C acquire/release logic for the OPT3001 sensor driver
* We fix the I2C acquire/release logic for the BMP280 sensor driver
* We fix the I2C acquire/release logic for the TMP007 sensor driver
* We fix the I2C acquire/release logic for the HDC1000 sensor driver
* We pass `-D<board>=1` instead of `-D<board>` at compile time. This is done to make this platform consistent with platform cc26x0-cc13x0.

@jsolderitsch any help with testing appreciated.

### CODE REVIEW

**Comment by nandroidj** [comment]
Hi,

 I want to add that this problem also occurs in the launchpad/cc1352r1 re-adapted to use it on the lpstk-cc1352r.

**Comment by jsolderitsch** [comment]
I am still seeing sensor support problems with the SimpleLink platform even with cc2650 and cc1350 SensorTags. Using PR 1290 to start testing from.

**Comment by g-oikonomou** [comment]
Changed base to branch `release-4.6`

**Comment by g-oikonomou** [comment]
Following a discussion with @alexstanoev, we are deferring this to v4.7. We have changed the base branch for this PR back to `develop`.

File: arch/platform/simplelink/cc13xx-cc26xx/sensortag/tmp-007-sensor.c [hunk 1]
```diff
@@ -256,14 +286,21 @@ read_data(uint16_t *local_tmp, uint16_t *obj_tmp)
   uint8_t status_data[] = { REG_STATUS };
   uint16_t status_value = 0;
 
+  if(i2c_acquire()) {
```
Comment (alexstanoev) [inline arch/platform/simplelink/cc13xx-cc26xx/sensortag/tmp-007-sensor.c:289]: Should be `if(!i2c_acquire()) {`

**Review by alexstanoev — CHANGES_REQUESTED** [review]

File: arch/platform/simplelink/cc13xx-cc26xx/sensortag/tmp-007-sensor.c (hunk 1 above)
Comment (g-oikonomou) [reply arch/platform/simplelink/cc13xx-cc26xx/sensortag/tmp-007-sensor.c]: Rebased over develop and addressed this.

**Review by alexstanoev — APPROVED** [review]

**Pull request merged**